src/ketu/async/sink.clj

```diff
@@ -205,6 +197,6 @@
   It shouldn't be necessary to call this function since by default the sink closes
   itself automatically when the input channel is closed."
   [state]
-  (wait-until-done-or-timeout! state)
   (abort-input! state)
+  (wait-until-done-or-timeout! state)
```

Comment:
The order of these calls is intentional. It first drains the producer channel or times out. Only then it does a hard abort. With this original order, you don't need `Thread/sleep` in the tests as well.

Reply:
Changed back to original order.